pyproject.toml

```diff
@@ -0,0 +1,59 @@
+[build-system]
+requires = ["setuptools>=64"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "pybop"
+version = "23.12"
+authors = [
+  {name = "The PyBOP Team"},
+]
+maintainers = [
+  {name = "The PyBOP Team"},
+]
+description = "Python Battery Optimisation and Parameterisation"
+readme = "README.md"
+license = { file = "LICENSE" }
+# https://pypi.org/classifiers/
+classifiers = []
+requires-python = ">=3.8, <=3.12"
```

Comment:
I am curious about the Python version bounds, because I copied this verbatim from `setup.py` – since PyBOP does not support Python 3.12 yet, this should have been `<3.12` rather than `<=3.12`. This makes it available, say, for Python 3.12.0, but not Python 3.12.1, so it's not a big deal altogether.

Please let me know if we should explicitly mark it as `<3.12` for now and add 3.12 support in the next release – though I think PyBaMM's own Python version constraint should ensure that functionality for PyBOP is as intended.

Reply:
Nice catch @agriyakhetarpal. Let's update this constraint to be `<3.12` and I'll open an issue to add support for `3.12`.